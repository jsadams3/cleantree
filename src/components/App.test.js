import React from 'react'

import { shallow } from 'enzyme'
import { App } from './'

const setHookState = (newState) =>
    jest.fn().mockImplementation(() => [newState, () => {}])

describe('App', () => {
    afterEach(() => {
        jest.clearAllMocks()
    })

    it('should render the component', () => {
        const wrapper = shallow(<App />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should render two inputs', () => {
        const wrapper = shallow(<App />)
        expect(wrapper.find('input')).toHaveLength(2)
    })

    it('should not render the Dashboard for initial state', () => {
        const wrapper = shallow(<App />)
        expect(wrapper.find('Dashboard').exists()).toBeFalsy()
    })

    it('should render the Dashboard if showRepos is true', () => {
        const wrapper = shallow(<App />)
        setHookState({
            inputValue: '',
            searchTerm: '',
            showRepos: true
        })
        expect(wrapper.find('Dashboard').exists())
    })
})
