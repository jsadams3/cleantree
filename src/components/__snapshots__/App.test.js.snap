// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component 1`] = `
<div
  className="app"
>
  <div
    className="search-container"
  >
    <input
      className="search-input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search for an organization"
      type="text"
      value=""
    />
    <input
      className="button search-button"
      id="search"
      onClick={[Function]}
      type="button"
      value="search"
    />
  </div>
</div>
`;
